ml/pom.xml

```diff
@@ -0,0 +1,346 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright (c) 2021, NVIDIA CORPORATION.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.nvidia</groupId>
+        <artifactId>rapids-4-spark-parent</artifactId>
+        <version>21.10.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>rapids-4-spark-ml_2.12</artifactId>
+    <name>RAPIDS Accelerator for Apache Spark ML</name>
+    <description>The RAPIDS cuML library for Apache Spark</description>
+    <version>21.10.0-SNAPSHOT</version>
+
+    <properties>
+        <java.version>1.8</java.version>
+        <spark.version>3.1.2</spark.version>
+         <scala.version>2.12.14</scala.version>
+        <test.java.home>${java.home}</test.java.home>
+        <spark.test.webdriver.chrome.driver></spark.test.webdriver.chrome.driver>
+        <spark.test.docker.keepContainer>false</spark.test.docker.keepContainer>
+        <spark.test.docker.removePulledImage>true</spark.test.docker.removePulledImage>
+        <CodeCacheSize>128m</CodeCacheSize>
+        <build.copyDependenciesPhase>none</build.copyDependenciesPhase>
+        <test.default.exclude.tags>org.apache.spark.tags.ChromeUITest</test.default.exclude.tags>
+        <test.exclude.tags></test.exclude.tags>
+        <test.include.tags></test.include.tags>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.scala-lang</groupId>
+            <artifactId>scala-library</artifactId>
+            <version>${scala.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-mllib_${scala.binary.version}</artifactId>
+            <version>${spark.version}</version>
+        </dependency>
+
+        <!-- Test -->
+        <dependency>
+            <groupId>org.scalatest</groupId>
+            <artifactId>scalatest_${scala.binary.version}</artifactId>
+            <version>3.2.9</version>
```

Comment:
we should inherit dependency versions from the parent profiles/dependency management

Reply:
Thanks for the suggestion, I agree. But I tried to use version from parent and the compilation will fail. 
Most of the depencies are copied from Spark ML module to make sure it can run under the ML framework.
Currently this module is like an independent module though it's in the project. We haven't cover all versions for Spark and for the initial drop, can we firstly make it compilable (though for specific versions)?